pennylane/devices/_qubit_device.py

```diff
@@ -1038,6 +1063,32 @@ def vn_entropy(self, wires, log_base):
         wires = wires.tolist()
         return qml.math.vn_entropy(state, indices=wires, c_dtype=self.C_DTYPE, base=log_base)
 
+    def vn_entanglement_entropy(self, wires0, wires1, log_base):
+        r"""Returns the Von Neumann entanglement entropy prior to measurement.
+        .. math::
+            S(\rho_A) = -\text{Tr}[\rho_A \log \rho_A] = -\text{Tr}[\rho_B \log \rho_B] = S(\rho_B)
+        Args:
+            wires0 (Sequence[int] or int): the wires of the first subsystem
+            wires1 (Sequence[int] or int): the wires of the second subsystem
+            log_base (float): Base for the logarithm.
+        Returns:
+            float: returns the Von Neumann entropy
+        """
+        try:
+            state = self.density_matrix(wires=self.wires)
+        except qml.QuantumFunctionError as e:  # pragma: no cover
```

Comment:
Huh, I never caught that. I'm not able to see where a QuantumFunctionError could be getting raised from. Nice, I'll change it to `NotImplementedError`.

Reply:
I don't quite understand what you mean. When would `QubitDevice.density_matrix` ever raise a `QuantumFunctionError` to be caught by the `except` block here? Additionally, if the general API is to raise a `QuantumFunctionError` for unsupported measurements, shouldn't you be raising a `QuantumFunctionError` instead of a `NotImplementedError` below?